### ISSUE TITLE
Does credential with no password make sense?

### ISSUE THREAD

**Issue by mslw** [datalad/datalad-next#438]
Nextcloud's public share links can be translated into WebDAV URLs like so ([Nextcloud docs](https://docs.nextcloud.com/server/19/user_manual/files/access_webdav.html#accessing-public-shares-over-webdav)): 
```https://example.com/nextcloud/public.php/webdav/...```

Required credentials are: share token as username, optional password as password.

I was making a dataset with addurls and uncurl special remote.

When there was no password in place (i.e. only username required), I got a prompt for `user_password` credential, and when I left password empty I got an error. However, when I entered a dummy password (`xxx`), it worked. I'm not sure if the error came from credential creation, or its subsequent usage, and didn't save the message.

Sorry for no reproducer (I can re-do and provide one, but that example dataset was throwaway) and an anecdotal nature of the report - does it make sense?

**Comment by adswa** [comment]
Somehwat unrelated: When I read this I was reminded of token-based access to GitHub, which in datalad-core (I believe) works when supplying the token, counterintuively, as the username and the password in the ``user_password`` prompt. So at least I've seen a ``user_password`` prompt working with a token before. 

Reading the documentation you linked, it sounds like the password does have a potential role as a "share password", though, right?

> in a WebDAV client, use the share token as username and the (optional) share password as password.

**Comment by mslw** [comment]
> Reading the documentation you linked, it sounds like the password does have a potential role as a "share password", though, right?
> 
> > in a WebDAV client, use the share token as username and the (optional) share password as password.

Yes, it does - but if the share has no password enabled, then there is no password. I'm not sure how exactly that translates into a request header, but the `curl` equivalent would be `curl -u user:password` vs `curl -u user`.

**Comment by mih** [comment]
Given the issue title my basic response is: yes, a credential with just a secret makes sense (it is a token).

However, I believe the issue is hinting at a (now) uncommon use of the credential *type* user_password -- anonymous login. Which makes sense to support.
